lib/test_queue/runner.rb

```diff
@@ -361,6 +372,46 @@ def distribute_queue
       end
     end
 
+    def check_quorum(remote_workers)
+      return unless quorum_failed?(remote_workers)
+      quorum_failed(remote_workers)
```

Comment:
`quorum_reached?` is problematic because it would have to return `true` before the timeout expires, but that feels incorrect. It could be two methods: `quorum_reached?` and `quorum_timeout_reached?`.